hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -831,8 +831,14 @@ def _annotate_args(
 ) -> Iterable[str]:
     arg_parameters: DefaultDict[str, Set[Any]] = defaultdict(set)
     for func in funcs:
-        for key, param in _get_params(func).items():
-            arg_parameters[key].add(param.annotation)
+        try:
+            params = _get_params(func, eval_str=True)
+        except Exception:
+            # don't add parameters if the annotations could not be evaluated
+            pass
```

Comment:
I think we need to handle this *within* the `_get_params` function, because an error due to `eval_str=True` will instead trigger the docstring-based fallback, and potentially lead to a wrong signature being returned!

We can get the signature as described above, and then discard any annotations that are strings at this stage.

Reply:
I've now had a closer look at the `_get_params` function. As it turns out, the fallbacks will be ignored anyway, because they don't set the `annotation` property of the `inspect.Parameter`. I've refactored the code to call the `get_signature` directly.